pkg/model/bootstrapscript.go

```diff
@@ -86,6 +90,40 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup, ps *kops.Egress
 			}
 			return ""
 		},
+
+		"ClusterSpec": func() (string, error) {
+			spec := make(map[string]interface{})
+			spec["cloudConfig"] = cs.CloudConfig
+			spec["docker"] = cs.Docker
+			spec["kubelet"] = cs.Kubelet
+			spec["kubeProxy"] = cs.KubeProxy
+
+			if ig.IsMaster() {
+				spec["kubeAPIServer"] = cs.KubeAPIServer
+				spec["kubeControllerManager"] = cs.KubeControllerManager
+				spec["kubeScheduler"] = cs.KubeScheduler
+				spec["masterKubelet"] = cs.MasterKubelet
+			}
+
+			content, err := yaml.Marshal(spec)
```

Comment:
I wonder if we should actually use the typed version i.e. take the spec & zero out the bits we don't want, and then use the existing API marshalling.  Then we could use this as the versioned API between kops & nodeup...

But I'm getting ahead of myself - this :+1: